integrationtests/tests/tests.rs

```diff
@@ -893,22 +948,23 @@ async fn runs_consensus_if_tx_submitted() -> Result<()> {
 #[tokio::test(flavor = "multi_thread")]
 async fn runs_consensus_if_new_block() -> Result<()> {
     test(2, |fed, user, bitcoin, _, _| async move {
+        let bitcoin = bitcoin.lock_exclusive().await;
         let peg_in_address = user.client.get_new_pegin_address(rng()).await;
-        bitcoin.mine_blocks(100);
-        let (proof, tx) = bitcoin.send_and_mine_block(&peg_in_address, Amount::from_sat(1000));
+        bitcoin.mine_blocks(100).await;
+        let (proof, tx) = bitcoin
+            .send_and_mine_block(&peg_in_address, Amount::from_sat(1000))
+            .await;
         fed.run_consensus_epochs(1).await;
 
-        // If epochs run before the blocks are mined, user won't be able to peg-in
-        join_all(vec![
```

Comment:
When writing integration tests for the future, is the general guidance to assume that another test could be running at the same time, so waiting on global state should be discouraged? (or take a global lock).

These changes are great but we should probably outline some guidelines so new tests don't replicate these issues

Reply:
See the PR description for explanation. I could put it somewhere in the repo itself, but the question is where exactly. Root docstring of the testing crate maybe?